app/src/main/java/com/github/se/orator/model/symblAi/AudioRecorder.kt

```diff
@@ -47,8 +49,11 @@ class AudioRecorder(
         AudioRecord(
             MediaRecorder.AudioSource.MIC, sampleRate, channelConfig, audioFormat, bufferSize)
 
-    audioFile = File(context.cacheDir, "audio_record.wav") // Recording in WAV format
-
+    // Save file to the appropriate directory based on offline mode
+    val saveDir = if (isOffline) context.filesDir else context.cacheDir
+    audioFile =
+        File(saveDir, "audio_record_${System.currentTimeMillis()}.wav") // Recording in WAV format
```

Comment:
Be careful since we are using .mp3 format in the offline mode!

Reply:
Are you sure it is a good idea to use np3 in your implementation ? As Online Mode already uses .wav files for this.